pkg/gateway/multiparts/kv_migrate_test.go

```diff
@@ -0,0 +1,69 @@
+package multiparts_test
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/gateway/multiparts"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+	"github.com/treeverse/lakefs/pkg/kv/postgres"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+func TestMigrate(t *testing.T) {
+	ctx := context.Background()
+	database, _ := testutil.GetDB(t, databaseURI)
+	kvStore := kvtest.MakeStoreByName(postgres.DriverName, databaseURI)(t, ctx)
+	defer kvStore.Close()
+	dbTracker := multiparts.NewDBTracker(database)
+
+	data := createData(t, ctx, dbTracker, 300)
+
+	buf := bytes.Buffer{}
+	err := multiparts.Migrate(ctx, database.Pool(), &buf)
+	require.NoError(t, err)
+
+	testutil.MustDo(t, "Import file", kv.Import(ctx, &buf, kvStore))
+	kvTracker := multiparts.NewTracker(kv.StoreMessage{Store: kvStore})
+	for _, entry := range data {
+		m, err := kvTracker.Get(ctx, entry.UploadID)
+		require.NoError(t, err)
+		require.Equal(t, *m, entry)
+	}
+}
+
+func createData(t *testing.T, ctx context.Context, tracker multiparts.Tracker, size int) []multiparts.MultipartUpload {
```

Comment:
createMigrateTestData? needs a better name as a global func in the scope of multiparts test package